components/arm/fake/fake.go

```diff
@@ -32,17 +41,48 @@ var fakeModelJSON []byte
 
 // AttrConfig is used for converting config attributes.
 type AttrConfig struct {
-	FailNew      bool   `json:"fail_new"`
-	FailValidate bool   `json:"fail_validate"`
-	ArmModel     string `json:"arm-model"`
+	ArmModel  string `json:"arm-model"`
+	ModelPath string `json:"model-path,omitempty"`
+}
+
+func modelFromName(model, name string) (referenceframe.Model, error) {
+	var custom referenceframe.Model
+	switch resource.ModelName(model) {
+	case xarm.ModelName6DOF.Name:
+		return xarm.Model(name, 6)
+	case xarm.ModelName7DOF.Name:
+		return xarm.Model(name, 7)
+	case ur.ModelName.Name:
+		return ur.Model(name)
+	case yahboom.ModelName.Name:
+		return yahboom.Model(name)
+	case eva.ModelName.Name:
+		return eva.Model(name)
+	case ModelName.Name:
+		return referenceframe.UnmarshalModelJSON(fakeModelJSON, name)
+	case "custom":
```

Comment:
I don't think this is quite what we want; it's mildly kludgy to require a magic string in an unrelated field to pass in a path. And it certainly shouldn't be returned as an error

Reply:
Good practice is to return either a good value and `nil`, or `nil` plus an error. Using the `error` field to return this sort of information like you're doing here makes program logic harder to follow for future devs